### ISSUE TITLE
Feature request: Add classifier options for deploy and install maven artifact

### ISSUE THREAD

**Issue by jmorille** [finn-no/maven-deploy#14]
the maven artifact was define with an optional properties named : classifier 
please add it as options.

http://maven.apache.org/pom.html
classifier: You may occasionally find a fifth element on the coordinate, and that is the classifier. We will visit the classifier later, but for now it suffices to know that those kinds of projects are displayed as groupId:artifactId:packaging:classifier:version.

The command line to deploy is:
http://maven.apache.org/plugins/maven-deploy-plugin/examples/deploying-with-classifiers.html
```shell
mvn org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy-file -Durl=http://localhost:8081/repomanager/ \
                                                                            -DrepositoryId=some.id \
                                                                            -Dfile=path-to-your-artifact-jar \
                                                                            -DpomFile=path-to-your-pom.xml \
                                                                            -Dclassifier=bin
```

**Comment by gregersrygg** [comment]
Sounds reasonable and should be easy to add :) Do you want to make a pull-request for it?

**Comment by jmorille** [comment]
To guarantee the parallel execution of maven deploy file, it could be interesting to respect the maven file convention for file temporary:

And modify the method loke that 
function archivePath () {
    var conf = getConfig();
    return path.join(conf.buildDir, conf.artifactId + '-' + pkg.version + (conf.classifier ? '-' + conf.classifier) + '.' + conf.type);
}

**Comment by gregersrygg** [comment]
The default finalName for Maven 3 is `${artifactId}-${version}`. The classifier seems to come from the maven-jar-plugin: http://maven.apache.org/pom.html

You can pass finalName to the config to override the `finalName`:

```javascript
maven.config({
    //...,
    "finalName"    : "{name}-{version}"
});
```

I'm don't think `filterConfig` allows to use values from the config (only pkg) when filtering the template-strings. Also it will be difficult to avoid the - when there is no classifier, but you can always hard-code it if you know when calling the config-method:

```javascript
maven.config({
    //...,
    "finalName"    : "{name}-{version}-bin"
});
```

**Comment by jmorille** [comment]
Without classifier is true.
But a classifier artifact not have this own pom but share the pom of the artifact without classifier
This one of the existing workaround of the maven paradigm of one project = one artifact. 

You code not working in multi threading causing by just a file name and the possibility of a work around change nothing.
The local file as no importance for mvn deploy or install because it rename it witthehe convention previously enonced. The file name as to be unique, Use an guid or a hash of the file content it really not matter. But not a static file name.

**Comment by gregersrygg** [comment]
Sorry, I don't understand what you are saying. Can you please rephrase and/or fix spelling errors so it's easier to read? :angel: 

Regarding unique names, is that part of the classifier, or a different feature? If so, please create another issue for that.

**Comment by gregersrygg** [comment]
Solved by #15
